packages/credential-providers/src/resolveAwsCliV2Region.ts

```diff
@@ -0,0 +1,63 @@
+import { MetadataService } from "@aws-sdk/ec2-metadata-service";
+import { loadSharedConfigFiles } from "@smithy/shared-ini-file-loader";
+
+interface ResolveRegionOptions {
+  defaultRegion?: string;
+  profile?: string;
+}
+
+/**
+ * Resolves the AWS region following AWS CLI V2 precedence order.
+ */
+export const resolveAwsCliV2Region = async ({
+  defaultRegion,
+  profile,
```

Comment:
this is going to be inconsistent with other providers since we do not automatically associate a client's profile with this resolver if it is used in the context of a client